pkg/netpol/eval/internal/k8s/adminnetpol.go

```diff
@@ -0,0 +1,348 @@
+func doesNamespacesFieldMatchPeer(namespaces *metav1.LabelSelector, peer Peer) (bool, error) {
+	if peer.PeerType() == IPBlockType {
+		return false, nil // namespaces does not select IPs
+	}
+	namespacesSelector, err := metav1.LabelSelectorAsSelector(namespaces)
+	if err != nil {
+		return false, err
+	}
+	return namespacesSelector.Matches(labels.Set(peer.GetPeerNamespace().Labels)), nil
+}
+
+// doesPodsFieldMatchPeer returns if the given NamespacedPod object matches the given peer
+// a NamespacedPod object contains both NamespaceSelector and PodSelector
+func doesPodsFieldMatchPeer(pods *apisv1a.NamespacedPod, peer Peer) (bool, error) {
+	if peer.PeerType() == IPBlockType {
+		return false, nil // pods does not select IPs
+	}
+	nsSelector, err := metav1.LabelSelectorAsSelector(&pods.NamespaceSelector)
+	if err != nil {
+		return false, err
+	}
+	podSelector, err := metav1.LabelSelectorAsSelector(&pods.PodSelector)
+	if err != nil {
+		return false, err
+	}
+	return nsSelector.Matches(labels.Set(peer.GetPeerNamespace().Labels)) && podSelector.Matches(labels.Set(peer.GetPeerPod().Labels)), nil
+}
+
+// why could not success yet with
+// using generics to avoid duplicates in following two funcs `egressRuleSelectsPeer` and `ingressRuleSelectsPeer`:
+// (same for updateConnsIfEgressRuleSelectsPeer and updateConnsIfIngressRuleSelectsPeer)
+//
+// according to https://tip.golang.org/doc/go1.18#generics :
+// "The Go compiler does not support accessing a struct field x.f where x is of type parameter type even if all types in the type
+// parameter’s type set have a field f. We may remove this restriction in a future release."
+// till GO 1.21 this restriction is not removed yet.
+// for example:
+// replacing "func egressRuleSelectsPeer(rulePeers []apisv1a.AdminNetworkPolicyEgressPeer, dst Peer) (bool, error)" and
+//           "func ingressRuleSelectsPeer(rulePeers []apisv1a.AdminNetworkPolicyIngressPeer, src Peer) (bool, error)"
+//
+// with a func using generics like this :
+//   "func xgressRuleSelectsPeer[T apisv1a.AdminNetworkPolicyEgressPeer | apisv1a.AdminNetworkPolicyIngressPeer](rulePeers []T,
+//                             dst Peer) (bool, error)"
+// will fail with errors such as :
+//                  "rulePeers[i].Namespaces undefined (type T has no field or method Namespaces)"
+//
+// a useful way to skip the errors is to define an interface with some Getter funcs to be implemented on the "inheriting types"
+// of the parameters.
+// but in this case : since our parameters are not of local types, we can not define new methods (like getters) on them;
+// not even with using aliases since then we'll need to copy values in calling funcs into the aliases and
+// this is not more efficient than current solutions.
+//
+// @todo: if GO is upgraded to a release that does not has this restriction on types with same fields, replace following two "duplicated"
+// funcs with one func that uses generics type
+
```

Comment:
move this to an issue, and remove comment from here

Reply:
done, new issue #425